server/sonar-server/src/main/java/org/sonar/server/permission/UserPermissionChanger.java

```diff
@@ -58,6 +62,9 @@ public boolean apply(DbSession dbSession, UserPermissionChange change) {
       default:
         throw new UnsupportedOperationException("Unsupported permission change: " + change.getOperation());
     }
+    if (SYSTEM_ADMIN.equals(change.getPermission())) {
+      dbClient.userDao().updateRootFlagFromPermissions(dbSession, change.getUserId().getLogin(), defaultOrganizationProvider.get().getUuid());
```

Comment:
I recommend that backend like DAO always works with technical keys but not functional keys. Here I suggest to use user id (`change.getUserId().getId()`).

Side-note: as you commented last week, `UserId` should indeed be renamed `UserRef`. I'd do that soon.